src/app/tree/checkbox-tree/terra-checkbox-tree.component.ts

```diff
@@ -87,4 +81,85 @@ export class TerraCheckboxTreeComponent extends TerraBaseTreeComponent
             }
         }
     }
+
+    private upRecursiveCheckParentLeafs(leaf:TerraLeafInterface):void
+    {
+        let allChildrenChecked:boolean = true;
+        let noChildChecked:boolean = true;
+        let isIndeterminate:boolean = false;
+
+        if(leaf.leafParent)
+        {
+            for(let subLeaf of leaf.leafParent.subLeafList)
+            {
+                allChildrenChecked = subLeaf.checkboxChecked && allChildrenChecked;
+                noChildChecked = !subLeaf.checkboxChecked && noChildChecked;
+                if(subLeaf.isIndeterminate)
+                {
+                    isIndeterminate = subLeaf.isIndeterminate;
+                }
+            }
+            if(allChildrenChecked)
+            {
+                leaf.leafParent.isIndeterminate = false;
+                leaf.leafParent.checkboxChecked = true;
+                this.upRecursiveCheckParentLeafs(leaf.leafParent);
+            }
+            else if(noChildChecked && isIndeterminate)
+            {
+                leaf.leafParent.checkboxChecked = null;
+                leaf.leafParent.isIndeterminate = true;
+                this.upRecursiveCheckParentLeafs(leaf.leafParent);
+            }
+            else if(noChildChecked)
+            {
+                leaf.leafParent.isIndeterminate = false;
+                leaf.leafParent.checkboxChecked = false;
+                this.upRecursiveCheckParentLeafs(leaf.leafParent);
+            }
+            else
+            {
+                leaf.leafParent.checkboxChecked = null;
+                leaf.leafParent.isIndeterminate = true;
+                if(leaf.leafParent.leafParent)
+                {
+                    this.recursiveSetIndeterminate(leaf.leafParent.leafParent);
+                }
+            }
+        }
+    }
+
+    private linkParentsToLeafList(leafList:Array<TerraLeafInterface>):void
+    {
+        for(let leaf of leafList)
+        {
+            if(leaf.subLeafList)
+            {
+                for(let subLeaf of leaf.subLeafList)
+                {
+                    subLeaf.leafParent = leaf;
+                    if(subLeaf.subLeafList)
+                    {
+                        this.linkParentsToLeafList(subLeaf.subLeafList);
+                    }
+                }
+            }
+        }
+    }
+
+    ngOnInit()
+    {
+        super.ngOnInit();
+        this.linkParentsToLeafList(this.inputLeafList);
+    }
+
+    private recursiveSetIndeterminate(leaf:TerraLeafInterface)
```

Comment:
Rückgabetyp und wäre wenn ich die Funktion irgendwo lese, klar in welche Richtung ich rekursiv gehe? 
Also childrens oder parents.